addons/package_hierarchy/tests/common.py

```diff
@@ -99,4 +105,7 @@ def create_quant(cls, product_id, location_id, qty, **kwargs):
             "quantity": qty,
         }
         vals.update(kwargs)
+        #Ensure quants are reserved in order of creation
+        if not "in_date" in vals:
+            vals.update({"in_date": Datetime.now() + timedelta(0, next(cls.quant_counter))})
```

Comment:
Could we avoid the conditional with [dict.setdefault](https://docs.python.org/3.10/library/stdtypes.html#dict.setdefault)? Some people think it's a bit obscure though.

```python
vals.setdefault('in_date', Datetime.now() + timedelta(0, next(cls.quant_counter))
```